src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -1540,14 +1552,16 @@ void ObjectMonitor::wait(jlong millis, bool interruptible, TRAPS) {
           ret = current->_ParkEvent->park(millis);
         }
       }
-
-      // were we externally suspended while we were waiting?
-      if (current->handle_special_suspend_equivalent_condition()) {
-        // TODO-FIXME: add -- if succ == current then succ = null.
-        current->java_suspend_self();
+      current->set_thread_state_fence(_thread_blocked_trans);
+      if (SafepointMechanism::should_process(current)) {
+        if (_succ == current) {
+            _succ = NULL;
+            OrderAccess::fence();
```

Comment:
My original comment is marked "fixed", but I don't see the new comment:

Please add a comment to this line:
OrderAccess::fence(); // always do a full fence when successor is cleared

Reply:
I fixed the one at line 982, it was not clear there were two from your comment :) Fixed